lib/middleware/autoUpdateMiddleware.ts

```diff
@@ -0,0 +1,100 @@
+// Update-notifier is CommonJS, so we need to use require
```

Comment:
This file contains the core of the auto update logic. It wasn't very difficult to do because we already had the utils to install deps. I abstracted them into the npm.ts file and leveraged them in here.